luigi_pipeline/tests/test_seqr_loading_tasks.py

```diff
@@ -49,3 +51,21 @@ def test_seqr_loading_validate_wgs_mismatch(self, mock_sample_type_stats):
         # Supposed to be WGS but we report as WES.
         mock_sample_type_stats.return_value = self._sample_type_stats_return_value(0, 0, True, 0, 0, True)
         self.assertRaises(SeqrValidationError, SeqrVCFToMTTask.validate_mt, None, '37', 'WES')
+
+    def test_derive_fields(self):
+        rsid = 'rs35471880'
+
+        mt = hl.import_vcf(TEST_DATA_MT_1KG)
+        mt = hl.split_multi(mt.filter_rows(mt.rsid == rsid))
+        mt = mt.annotate_rows(**VEP_DATA[rsid])
+        mt = SeqrVCFToMTTask.derive_fields(mt, 'VARIANTS')
+
+        obj = mt.rows().collect()[0]
+
+        # Cannot do a nested compare because of nested objects, so do one by one.
```

Comment:
I thought [assertEqual](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual) could handle that? Is it because the objects are Hail structs?

Reply:
Yes, I should put in nested hail objects. It looks like the hail comparator isn't set. I tried getting around that by comparing against converted json sorted on field names, as well as converting to dicts recursively, but didn't get it to work in the time box.